spring-integration-core/src/main/java/org/springframework/integration/support/management/IntegrationManagementConfigurer.java

```diff
@@ -210,7 +212,20 @@ public void afterSingletonsInstantiated() {
 		Assert.state(this.applicationContext != null, "'applicationContext' must not be null");
 		Assert.state(MANAGEMENT_CONFIGURER_NAME.equals(this.beanName), getClass().getSimpleName()
 				+ " bean name must be " + MANAGEMENT_CONFIGURER_NAME);
-		this.metricsCaptor = MetricsCaptorLoader.loadCaptor(this.applicationContext);
+		ClassLoader classLoader = IntegrationManagementConfigurer.class.getClassLoader();
+		if (ClassUtils.isPresent("io.micrometer.core.instrument.MeterRegistry",
+				classLoader)) {
+			try {
+				Class<?> captor = ClassUtils.forName(
+						"org.springframework.integration.support.management.micrometer.MicrometerMetricsCaptor",
+						classLoader);
+				Method method = captor.getDeclaredMethod("loadCaptor", ApplicationContext.class);
+				this.metricsCaptor = (MetricsCaptor) method.invoke(null, applicationContext);
```

Comment:
Why do we need all these reflection kung-fu?
Seems for me since `MicrometerMetricsCaptor.loadCaptor()` is `static` we are good around proper class loading.

Thank

Reply:
I tried that and, even with a fully qualified class name (no import), I got a CNFE in a test app that didn't have Micrometer on the CP.

Reply:
Must have been pilot error - works now.

Reply:
Actually, no; my memory is bad; reflection is needed to "avoid" the package tangle between `management` and `micrometer`.

The alternative would be to move all the classes in `micrometer` up to `management`. 

WDYT?

Reply:
Good point!

Let's add a comment here on these lines to explain the reason for the reflection and that's all!

Please, also fix `this.` prefix for this `applicationContext` property.